### DISCUSSION TITLE
.activateAR() on Android fails?

### DISCUSSION THREAD

**Discussion by richardrevesz** [google/model-viewer#2410] [Q&A]
Hi,
I am trying to auto trigger the AR mode from js with:
`modelViewer.activateAR();
`
It works fine on my iPad (auto generated USDZ), but the AR mode on Android seems to fail and the AR button gets hidden.

Any idea about this?

**Comment by elalish** [comment]
Any console logs? And do you have a link where I can repro?

**Reply by richardrevesz** [reply]
@elalish please find it here: https://www.aronplatform.com/canvas-print-ar-demo/
you need to upload an image, select a frame size and apply crop.

**Reply by elalish** [reply]
Here's your helpful console error message: `DOMException: Failed to execute 'requestSession' on 'XRSystem': The requested session requires user activation.`

Where are you calling `activateAR()`, and why not just use our AR button? The error message says it all; you can't start WebXR without a user clicking something.

**Reply by richardrevesz** [reply]
I reorganized it a bit, so it is explicitly on button click, but still get the same error.

`document.getElementById('cropbutton').onclick = async function(){ 
arReady = false;
await until (_ => arReady == true);
modelViewerTexture.activateAR() 
};
`
(I set arReady to true when it's ready)

I want to streamline the user experience, so reduce the necessary clicks as much as possible.
Any suggestion how to achieve it?

**Reply by elalish** [reply]
As I said before, this isn't a click you can get around. Your async function is waiting until that click is stale.

**Reply by richardrevesz** [reply]
Oh, I see. What a pity. Thank you for the clarification!

**Comment by david-rhodes** [comment]
@elalish Is there a reason why you can `activateAR` with a scene-viewer or QuickLook intent without user interaction (via JS), but we cannot do so with model-viewer? Is this purely a limitation of WebXR?

Thank you!

**Reply by elalish** [reply]
I believe Chrome takes the security a bit more seriously with WebXR since it remains in the browser and so has more access to data.
